src/app/(mobile-ui)/wallet/page.tsx

```diff
@@ -39,7 +40,9 @@ const WalletDetailsPage = () => {
 
                 <Card shadowSize="4" className="w-full rounded-md py-10">
                     <Card.Content className="flex h-full flex-row items-center justify-center">
-                        <div className="text-5xl">{'$ 420.69'}</div>
+                        <div className="text-5xl">
+                            $ {Number(formatUnits(selectedWallet?.balance ?? 0n, 6)).toFixed(2)}
```

Comment:
formatUnit handles part of that, it will return the USD(C) amount. Unless the balance is too big, in that case you are right and we would have problems with scientific notation.
I have this task https://www.notion.so/peanutprotocol/Bug-X-chain-from-amount-formatted-wrongly-to-scientific-notation-141838117579807abd12ee396cbfee97?pvs=4 to deal with that